frontend/src/concepts/modelRegistry/utils.ts

```diff
@@ -113,3 +114,80 @@ export const filterArchiveModels = (registeredModels: RegisteredModel[]): Regist
 
 export const filterLiveModels = (registeredModels: RegisteredModel[]): RegisteredModel[] =>
   registeredModels.filter((rm) => rm.state === ModelState.LIVE);
+
+/**
+ * Converts model source properties to catalog parameters
+ * @param modelSourceKind - The kind of model source
+ * @param modelSourceClass - The class of model source (maps to catalog source name)
+ * @param modelSourceGroup - The group of model source (maps to catalog repository name)
+ * @param modelSourceName - The name of model source (maps to catalog model name)
+ * @param modelSourceId - The ID of model source (maps to catalog model tag)
+ * @returns CatalogModelDetailsParams object or null if not a catalog source
+ */
+export const modelSourcePropertiesToCatalogParams = (
+  modelSourceKind?: string,
+  modelSourceClass?: string,
+  modelSourceGroup?: string,
+  modelSourceName?: string,
+  modelSourceId?: string,
+): CatalogModelDetailsParams | null => {
+  if (modelSourceKind !== ModelSourceKind.CATALOG) {
+    return null;
+  }
+
+  return {
+    sourceName: modelSourceClass || '',
+    repositoryName: modelSourceGroup || 'default',
+    modelName: modelSourceName || '',
+    tag: modelSourceId || '',
+  };
+};
+
+/**
+ * Converts catalog parameters to model source properties
+ * @param params - The catalog model details parameters
+ * @returns Object containing model source properties
+ */
+export const catalogParamsToModelSourceProperties = (
+  params: CatalogModelDetailsParams,
+): {
+  modelSourceKind: ModelSourceKind;
+  modelSourceClass?: string;
+  modelSourceGroup?: string;
+  modelSourceName?: string;
+  modelSourceId?: string;
+} => ({
+  modelSourceKind: ModelSourceKind.CATALOG,
+  modelSourceClass: params.sourceName,
+  modelSourceGroup: params.repositoryName,
+  modelSourceName: params.modelName,
+  modelSourceId: params.tag,
+});
+
+/**
+ * Converts model source properties to pipeline run reference
+ * @param modelSourceKind - The kind of model source
+ * @param modelSourceGroup - The group of model source (project name)
+ * @param modelSourceId - The ID of model source (run ID)
+ * @param modelSourceName - The name of model source (run name)
+ * @returns PipelineRunRef object or null if not a DSP source
+ */
+export const modelSourcePropertiesToPipelineRunRef = (
+  modelSourceKind?: string,
+  modelSourceGroup?: string,
+  modelSourceId?: string,
+  modelSourceName?: string,
+): { project: string; runId: string; runName: string } | null => {
+  if (modelSourceKind !== ModelSourceKind.DSP) {
+    // For backward compatibility, also check if the fields exist without explicit DSP kind
+    if (!modelSourceGroup || !modelSourceId || !modelSourceName) {
+      return null;
+    }
+  }
```

Comment:
Backwards compatibility doesn't make sense here - if these properties exist they'll either exist with the right kind, or we can ignore them. This can just be one `if` that requires all of these, and that'll also narrow the type so you don't have to use `|| ''` below.

```ts
if (
  modelSourceKind !== ModelSourceKind.DSP ||
  !modelSourceGroup ||
  !modelSourceId ||
  !modelSourceName
) {
  return null;
}
```